lib/opencv_nx.ex

```diff
@@ -7,82 +7,74 @@ defmodule OpenCV.Nx do
   list.
   """
 
+  @compile {:no_warn_undefined, Nx}
+
   @doc """
-  Transform an OpenCV.Mat `#Reference` to `#Nx.tensor`
+  Transform an OpenCV.Mat `#Reference` to `Nx.tensor`.
+
+  The resulting tensor is in the shape `{width, height, channels}`.
+
   ### Example
+
   ```elixir
-  {:ok, mat} = OpenCV.imread("/path/to/exist/img.png")
-  nx_tensor = OpenCV.Nx.to_nx(mat)
-   #Nx.Tensor<
-      u8[1080][1920][3]
-      [[ ... pixel data ... ]]
-   >
+  iex> {:ok, mat} = OpenCV.imread("/path/to/exist/img.png")
+  iex> nx_tensor = OpenCV.Nx.to_nx(mat)
+  ...> #Nx.Tensor<
+  ...>    u8[1080][1920][3]
+  ...>    [[ ... pixel data ... ]]
+  ...> >
   ```
   """
   @doc namespace: :external
   @spec to_mat(reference()) :: {:ok, reference()} | {:error, String.t()}
   def to_nx(mat) do
-    if Code.ensure_loaded?(Nx) do
-      {:ok, mat_type} = OpenCV.Mat.type(mat)
-      {:ok, mat_shape} = OpenCV.Mat.shape(mat)
+    {:ok, mat_type} = OpenCV.Mat.type(mat)
+    {:ok, mat_shape} = OpenCV.Mat.shape(mat)
```

Comment:
You should consider having many of those functions raise, depending on the type of error, especially argument errors. We typically return tuples for semantic errors. For example, `File.read(...)` returns a `{:error, ...}` if a string is given but it does not exist. However, it will raise if you give it an integer. :)

Reply:
Oh yes, definitely need to write more specs or guards. The current status is like, letting it crash instead of failing fast or something else.

I'm also considering generating specs using the python script. It should be able to do that for, I don't know, maybe 50% of these functions.

I'll add this to the todo list!